Show how to compute profile for an image along polyline, remove unnecessary utils.py

gena opened the pull request:
... now sure is `geometry/` is a proper name, maybe `image/` ?

gena committed:
embed image in notebook

gena committed:
embed images, remove unnecessary python code

gena committed:
embed image

gena committed:
remove duplicate

gena committed:
add code to compute profile for a given image and polyling

tylere commented on examples/geometry/image_profile.ipynb:
```diff
@@ -0,0 +1,128 @@
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 40,
```
The cells appear to have been run out of order.
If you are saving a notebook file with outputs, be sure to reset the kernel, clear the outputs, and run all the cells before saving and committing the file.

gena replied:
Yes, this sounds better

tylere commented on examples/geometry/image_profile.ipynb:
```diff
@@ -0,0 +1,128 @@
+    "         [-114.3616, 36.4220],\n",
+    "         [-114.3514, 36.4203],\n",
+    "         [-114.3419, 36.4199],\n",
+    "         [-114.3116, 36.4228]]);\n",
```
Unnecessary semicolon.

gena replied:
Thanks, copy/pasting from JS, fixed!

tylere commented on examples/geometry/image_profile.ipynb:
```diff
@@ -0,0 +1,128 @@
+    "def reduceImageProfile(image, line, reducer, scale):\n",
+    "    length = line.length()\n",
+    "    distances = ee.List.sequence(0, length, scale)\n",
+    "    lines = line.cutLines(distances).geometries();\n",
```
Unnecessary semicolon.

gena replied:
fixed

tylere commented on examples/geometry/image_profile.ipynb:
```diff
@@ -0,0 +1,128 @@
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
```
The reduceImageProfile function seems to be the most important part of this example. This could be improved by adding notebook annotations that explain what the function is doing.

tylere commented on examples/export/utils.py:
```diff
@@ -1,31 +0,0 @@
-import urllib, io, os
```
This deletion appears unrelated to the PR. Could you either move this to a separate PR or expand the description of this PR?

gena replied:
extended descriptions, utils.py is not needed anymore because if embed=True

tylere replied:
ok, makes sense.

tylere commented on examples/getting-started/display-static-map.ipynb:
```diff
@@ -74,7 +72,7 @@
     "url = ee.Image(\"USGS/SRTMGL1_003\").getThumbUrl({'min':0, 'max':3000})\n",
     "# Display the image in the notebook.\n",
-    "Image(url=url)"
+    "Image(url=url, embed=True, format='png')"
```
This file appears unrelated to the PR. Could you move this to a separate PR, or expand the description of this PR if you think it is related?

tylere commented on examples/export/export-image-gdrive.ipynb:
```diff
@@ -2,13 +2,13 @@
   {
    "cell_type": "code",
-   "execution_count": 1,
+   "execution_count": 7,
```
The cells appear to have been run out of order.
If you are saving a notebook file with outputs, be sure to reset the kernel, clear the outputs, and run all the cells before saving and committing the file.

gena committed:
Processing review comments

gena commented:
I've reorganized notebook a bit:
* added a short description at the beginning
* added documentation for reduceImageProfile()
* moved the actual example to the end
